app.js

```diff
@@ -227,6 +227,27 @@ function App (el) {
     })
   })
 
+  self.on('executeCommand', function (commandStr) {
+    var words = commandStr.split(' ')
+    var command = words[0].substring(1, words[0].length).toLowerCase()
+
+    switch (command) {
+      case 'join':
+        words.shift()
+        var channel = words.join(' ')
+        self.emit('addChannel', channel)
+        break
+      case 'wc':
+      case 'part':
+      case 'leave':
+        self.emit('leaveChannel', self.data.activeChannel.name)
+        break
+      default:
+        console.log('Unrecognized command')
+        break
```

Comment:
@bcomnes this looks great! one thought -- maybe if the command is unrecognized it just passes through as a message instead?